apiserver/logsink.go

```diff
@@ -172,11 +215,41 @@ func (h *logSinkHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		// formatted simple error.
 		h.sendError(socket, req, nil)
 
-		logCh := h.receiveLogs(socket)
+		// Older versions did not respond to ping control messages, so don't try.
+		doPingPong := endpointVersion > 0
```

Comment:
I'm not sure we need this variable, given you use "endpointVersion" passed to the other function, just inline this to the "if doPingPong" check.